apis/projectcontour/v1/httpproxy.go

```diff
@@ -616,6 +620,33 @@ type HTTPRequestRedirectPolicy struct {
 	Prefix *string `json:"prefix,omitempty"`
 }
 
+type HTTPInternalRedirectPolicy struct {
+	// MaxInternalRedirects An internal redirect is not handled, unless the number of previous internal
+	// redirects that a downstream request has encountered is lower than this value.
+	// +optional
+	MaxInternalRedirects uint32 `json:"maxInternalRedirects,omitempty"`
+
+	// RedirectResponseCodes If unspecified, only 302 will be treated as internal redirect.
+	// Only 301, 302, 303, 307 and 308 are valid values.
+	// +optional
```

Comment:
I think we're missing validation (via kubebuilder or otherwise in the httpproxy processor) that valid status codes hasve been set here

Reply:
According to their documentation, Envoy simply ignores other status code, so no validation is "required", but maybe we should warn if an other status code is used ?

Reply:
We use kubebuilder validation for this elsewhere (https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/httpproxy.go#L596) so I think that would work here as well.

Reply:
`type RetryOn string` is not exactly a type alias, but a new type. When creating the DAG representation, the `RetryOn` slice must be explicitly converted to a `string[]` by creating a new `slice` and populating it with `string(RetryOn)` values.

kubebuilder does not support type alias (`type RedirectResponseCode = uint32`), so I will have to handle the conversion explicitly too.